In [1]:
import pandas as pd
df = pd.read_csv("MetaPosts.csv")



Análise das colunas

In [35]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 583 entries, 0 to 582
Data columns (total 24 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   rede_social           583 non-null    object 
 1   marca                 583 non-null    object 
 2   ano                   583 non-null    int64  
 3   mes                   583 non-null    int64  
 4   data_hora_criacao     583 non-null    object 
 5   dia_da_semana         583 non-null    object 
 6   link                  583 non-null    object 
 7   url_imagem            583 non-null    object 
 8   legenda               583 non-null    object 
 9   tipo                  583 non-null    object 
 10  alcance_organico      292 non-null    float64
 11  alcance_pago          292 non-null    float64
 12  alcance_total         583 non-null    int64  
 13  curtidas_e_reacoes    583 non-null    int64  
 14  comentarios           583 non-null    int64  
 15  compartilhamentos     5

Tranformar a taxa de engajamento para float para conseguir fazer os cálculos com ela

In [2]:

def obj_to_float(x):
    tranforma = float(x.replace(' ','')
                .replace(',', '.'))
    return tranforma

df['taxa_de_engajamento'] = df['taxa_de_engajamento'].apply(obj_to_float)


In [37]:
df.duplicated().sum()

np.int64(0)

Não tem nenhuma linha duplicada

Quantidade dos valores nulo de cada variável

In [7]:
df.isnull().sum()

rede_social               0
marca                     0
ano                       0
mes                       0
data_hora_criacao         0
dia_da_semana             0
link                      0
url_imagem                0
legenda                   0
tipo                      0
alcance_organico        291
alcance_pago            291
alcance_total             0
curtidas_e_reacoes        0
comentarios               0
compartilhamentos         0
cliques                   0
salvos                    0
views                   290
impressoes_organicas      0
impressoes_pagas          0
impressoes_totais         0
envolvimento              0
taxa_de_engajamento       0
dtype: int64

Quase metade dos dados de views, alcance orgânico e pago são nulos

In [23]:
df["salvos"].value_counts()

salvos
0      369
1       70
2       40
3       34
5       18
4       14
6        8
7        6
8        5
14       2
9        2
16       2
29       1
174      1
40       1
17       1
109      1
25       1
28       1
23       1
19       1
114      1
56       1
13       1
18       1
Name: count, dtype: int64

Grande número de 0

In [3]:
df["alcance_pago"].value_counts()

alcance_pago
0.0    292
Name: count, dtype: int64

Metade dos dados de alcance pago são nulos e a outra apenas 0.0

In [4]:
df["views"].value_counts()

views
0.0        292
22614.0      1
Name: count, dtype: int64

Apenas uma linha das views é diferente de 0

In [5]:
df["impressoes_pagas"].value_counts()

impressoes_pagas
0         582
144817      1
Name: count, dtype: int64

Apenas uma linha das impressões pagas é diferente de 0

In [6]:
df.describe()

,ano,mes,alcance_organico,alcance_pago,alcance_total,curtidas_e_reacoes,comentarios,compartilhamentos,cliques,salvos,views,impressoes_organicas,impressoes_pagas,impressoes_totais,envolvimento,taxa_de_engajamento
count,583.000000,583.000000,292.000000,292.0,583.000000,583.000000,583.000000,583.000000,583.000000,583.000000,293.000000,583.000000,583.000000,583.000000,583.000000,583.000000
mean,2025.161235,6.708405,2492.815068,0.0,1600.878216,65.962264,3.471698,12.149228,9.881647,2.180103,77.180887,3053.476844,248.399657,3303.778731,93.644940,0.291930
std,0.640613,3.405178,2436.415871,0.0,4410.060534,104.281218,7.318160,40.969606,52.619245,10.504302,1321.123986,5173.343744,5997.707322,7860.052074,160.391175,1.302824
min,2024.000000,1.000000,348.000000,0.0,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,2.000000,0.000000,0.000000,0.000000,0.000000
25%,2025.000000,4.000000,1289.000000,0.0,167.500000,8.000000,0.000000,0.000000,0.000000,0.000000,0.000000,308.500000,0.000000,310.000000,12.000000,0.038462
50%,2025.000000,6.000000,1965.000000,0.0,841.000000,30.000000,1.000000,2.000000,0.000000,0.000000,0.000000,1367.000000,0.000000,1369.000000,41.000000,0.057971
75%,2026.000000,10.000000,2847.000000,0.0,2068.000000,72.500000,3.000000,10.000000,3.000000,1.000000,0.000000,3708.000000,0.000000,3732.000000,97.500000,0.086552
max,2026.000000,12.000000,22812.000000,0.0,91421.000000,1020.000000,71.000000,587.000000,1004.000000,174.000000,22614.000000,54751.000000,144817.000000,145970.000000,1826.000000,23.500000


Impressões Totais
Desvio padrão é alto, conclui-se que tem uma alta disperção
Grande diferença entre a média e a mediana assim como grande diferença da média para o maior número, mostra que tem mais dados menores e que os poucos maiores elevam a média 
Maior número é muito maior que qualquer outro quartil, média ou mediana(outlier)

Alcance Total
Desvio padrão alto, indicando disperção
Média sendo quase o dobro da mediana, indica mais valores baixos e menos valores alto que elevam a média
Maior número é muito maior que qualquer outro quartil, média ou mediana(outlier)

Curtidas e Reações
Desvio padrão é alto, conclui-se que tem uma alta disperção
Média sendo mais que o dobro da mediana, indica mais valores baixos e menos valores alto que elevam a média
Maior número é muito maior que qualquer outro quartil, média ou mediana(outlier)




Mudar os nulos para 0.0, para ficarem iguais aos outros(teria que fazer strip pois o 0.0 quando substituido é visto como diferente do que já está lá)

In [52]:
df['views'] = df['views'].fillna('0.0')
df['alcance_pago'] = df['alcance_pago'].fillna('0.0')

In [7]:
df.isnull().sum()

rede_social               0
marca                     0
ano                       0
mes                       0
data_hora_criacao         0
dia_da_semana             0
link                      0
url_imagem                0
legenda                   0
tipo                      0
alcance_organico        291
alcance_pago            291
alcance_total             0
curtidas_e_reacoes        0
comentarios               0
compartilhamentos         0
cliques                   0
salvos                    0
views                   290
impressoes_organicas      0
impressoes_pagas          0
impressoes_totais         0
envolvimento              0
taxa_de_engajamento       0
dtype: int64

In [ ]:
df.drop(columns=['views', 'impressoes_pagas', 'alcance_pago'])

In [12]:
df.head()

,rede_social,marca,ano,mes,data_hora_criacao,dia_da_semana,link,url_imagem,legenda,tipo,...,comentarios,compartilhamentos,cliques,salvos,views,impressoes_organicas,impressoes_pagas,impressoes_totais,envolvimento,taxa_de_engajamento
0,Instagram,B Braun,2026,7,03/07/2026 12:01:47,Sexta-Feira,https://www.instagram.com/p/DaVHjZqjF3u,https://storage.googleapis.com/brv-valhalla-da...,E se você pudesse conhecer a tecnologia OMNI a...,reels,...,5,27,0,1,0.0,2571,0,2571,89,0.052016
1,Facebook,B Braun,2026,7,03/07/2026 12:01:06,Sexta-Feira,https://www.facebook.com/reel/2536486383534747/,https://storage.googleapis.com/brv-valhalla-da...,E se você pudesse conhecer a tecnologia OMNI a...,reels,...,0,0,0,0,NaN,74,0,75,4,0.057971
2,Facebook,B Braun,2026,7,02/07/2026 13:28:01,Quinta-Feira,https://www.facebook.com/reel/1610329023941411/,https://storage.googleapis.com/brv-valhalla-da...,Um hospital é movido por pessoas. Pessoas que ...,reels,...,0,0,1,0,NaN,219,0,219,5,0.023810
3,Instagram,B Braun,2026,7,02/07/2026 13:26:15,Quinta-Feira,https://www.instagram.com/p/DaSsbGZj7Oc,https://storage.googleapis.com/brv-valhalla-da...,Um hospital é movido por pessoas. Pessoas que ...,reels,...,0,6,0,0,0.0,1449,0,1449,64,0.061896
4,Instagram,B Braun,2026,6,29/06/2026 12:00:16,Segunda-Feira,https://www.instagram.com/p/DaK0SiaiMVL,https://storage.googleapis.com/brv-valhalla-da...,Higienizar as mãos é um protocolo de segurança...,carrossel,...,0,6,0,1,0.0,1536,0,1536,43,0.058266


Agrupamento por ano, mês e rede social

In [ ]:
metricas = df.groupby(["ano", "mes", "rede_social"])[["alcance_total","curtidas_e_reacoes","impressoes_totais"]].sum()
metricas

In [ ]:
porcentagem_taxa_engajamento = (df.groupby(["ano", "mes", "rede_social"])['taxa_de_engajamento'].mean())
porcentagem_taxa_engajamento

crescimento anual


In [ ]:
crescimento_ano = metricas_ano.pct_change()
crescimento_ano.round(2)

taxa metrica anual

In [ ]:
taxa_engajamento_ano = (df.groupby("ano")['taxa_de_engajamento'].mean())
taxa_engajamento_ano = metricas_ano.pct_change() 
taxa_engajamento_ano

Meses em comum entre 2025 e 2026(meses 1,2,3,4,5,6,7)

In [24]:

a25_a26 = df[df['ano'].isin([2025, 2026])]
mes_comum = a25_a26[a25_a26['mes'].isin([1,2,3,4,5,6,7])]
metricas_mes = mes_comum.groupby(['mes', "ano"])[["alcance_organico","alcance_total", "curtidas_e_reacoes", "comentarios","compartilhamentos", "cliques","salvos","views","impressoes_totais", "impressoes_organicas", "impressoes_pagas", "envolvimento"]].sum()
metricas_mes

alcance_organico  alcance_total  curtidas_e_reacoes  comentarios  \
mes ano                                                                      
1   2025           25636.0          25902                1661          137   
    2026            9077.0          11067                 428           12   
2   2025           13444.0          13655                 547           25   
    2026            9603.0          12543                 688           49   
3   2025           41992.0          42462                2401          143   
    2026           46287.0          54032                3047          103   
4   2025           20938.0          21295                 955           36   
    2026           47991.0          63472                2510          114   
5   2025           54894.0          58558                2569          101   
    2026           69038.0         105085                4187          161   
6   2025           26351.0          28551                1147           71   
    2026           25558.0          33422                1556           88   
7   2025           20640.0          23723                 962           60   
    2026            2745.0           3024                 122            5   

          compartilhamentos  cliques  salvos  views  impressoes_totais  \
mes ano                                                                  
1   2025                169      376      28    0.0              53467   
    2026                 34       33       3    0.0              16203   
2   2025                 88       20      17    0.0              22980   
    2026                101       66      16    0.0              20911   
3   2025                354      374      47    0.0              74435   
    2026                460      411      53    0.0              86798   
4   2025                154       41      22    0.0              39276   
    2026                847      620     232    0.0             127507   
5   2025                561      164      38    0.0              97577   
    2026                919     1148     168    0.0             163275   
6   2025                203       73      23    0.0              51165   
    2026                224      475      51    0.0              59422   
7   2025                170      204      22    0.0              51127   
    2026                 33        1       1    0.0               4314   

          impressoes_organicas  impressoes_pagas  envolvimento  
mes ano                                                         
1   2025                 53462                 0          2371  
    2026                 16202                 0           510  
2   2025                 22975                 0           697  
    2026                 20906                 0           920  
3   2025                 74434                 0          3319  
    2026                 86791                 0          4074  
4   2025                 39274                 0          1208  
    2026                127506                 0          4323  
5   2025                 97570                 0          3433  
    2026                163268                 0          6583  
6   2025                 51162                 0          1517  
    2026                 59421                 0          2394  
7   2025                 51125                 0          1418  
    2026                  4313                 0           162

um ano de cada vez

In [ ]:
a25 = df[df['ano'] == 2025]
mes_comum = a25[a25['mes'].isin([1,2,3,4,5,6,7])]
metricas_a25 = mes_comum.groupby('mes')[["alcance_organico","alcance_total", "curtidas_e_reacoes", "comentarios","compartilhamentos", "cliques","salvos","views","impressoes_totais", "impressoes_organicas", "impressoes_pagas", "envolvimento"]].sum()
metricas_a25

In [ ]:
a26 = df[df['ano'] == 2026]
mes_comum = a26[a26['mes'].isin([1,2,3,4,5,6,7])]
metricas_a26 = mes_comum.groupby('mes')[["alcance_organico","alcance_total", "curtidas_e_reacoes", "comentarios","compartilhamentos", "cliques","salvos","views","impressoes_totais", "impressoes_organicas", "impressoes_pagas", "envolvimento"]].sum()
metricas_a26

Taxa de varição das métricas nos meses em comum

In [ ]:
taxa_de_variacao = ((metricas_a26 - metricas_a25) / metricas_a25 * 100)
taxa_de_variacao.round(2)

Média da taxa de variação entre os meses em comum

In [ ]:
media_taxa_de_variacao = taxa_de_variacao.mean().round(2).astype(str) + " %"
media_taxa_de_variacao

Tranformar a taxa de engajamento para fazer o calcúlo

In [ ]:
df["taxa_de_engajamento"] = (
    df["taxa_de_engajamento"]
    .str.replace(",", ".")
    .astype(float)
)

In [ ]:
taxa_2025 = (
    a25[a25["mes"].isin([1,2,3,4,5,6,7])]
    .groupby("mes")["taxa_de_engajamento"]
    .mean()
)*100
taxa_2025

In [ ]:
taxa_2026 = (
    a26[a26["mes"].isin([1,2,3,4,5,6,7])]
    .groupby("mes")["taxa_de_engajamento"]
    .mean()
)*100
taxa_2026

Taxa de variação da taxa de engajamento

In [ ]:
crescimento_taxa = (
    (taxa_2026 - taxa_2025)
    / taxa_2025
)*100
crescimento_taxa.round(2)

Média da taxa de variação da taxa de engajamento

In [ ]:
taxa_previsao = crescimento_taxa.mean().round(2)
taxa_previsao.astype(str) + " %"

Gráficos

In [ ]:
import matplotlib.pyplot as plt


Meses em comum 25-26

Alcance pago não tem dados suficientes

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["alcance_organico"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["alcance_organico"],
    marker="o",
    label="2026"
)

plt.title("Alcance Orgânico - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Alcance Orgânico")

plt.legend()
plt.grid()

plt.show()

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["alcance_total"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["alcance_total"],
    marker="o",
    label="2026"
)

plt.title("Alcance Total - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Alcance total")

plt.legend()
plt.grid()

plt.show()

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["curtidas_e_reacoes"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["curtidas_e_reacoes"],
    marker="o",
    label="2026"
)

plt.title("Curtidas e Reações - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de Curtidas e Reações")

plt.legend()
plt.grid()

plt.show()

In [ ]:
plt.plot(
    metricas_a25.index,
    metricas_a25["comentarios"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["comentarios"],
    marker="o",
    label="2026"
)

plt.title("Comentários - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de comentários")

plt.legend()
plt.grid()

plt.show()

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["compartilhamentos"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["compartilhamentos"],
    marker="o",
    label="2026"
)

plt.title("Compartilhamentos - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de Compartilhamentos")

plt.legend()
plt.grid()

plt.show()

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["cliques"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["cliques"],
    marker="o",
    label="2026"
)

plt.title("Cliques - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de cliques")

plt.legend()
plt.grid()

plt.show()

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["salvos"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["salvos"],
    marker="o",
    label="2026"
)

plt.title("Salvos - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de Salvos")

plt.legend()
plt.grid()

plt.show()

As views são nulas exceto por um valor, então não foi considerada

Como não haviam impressões pagas o gráfico das impressões orgânicas ficaria muito parecido com o de impressões totais

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["impressoes_totais"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["impressoes_totais"],
    marker="o",
    label="2026"
)

plt.title("Impressões Totais - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de Impressões")

plt.legend()
plt.grid()

plt.show()

Envolvimento

In [ ]:

plt.plot(
    metricas_a25.index,
    metricas_a25["envolvimento"],
    marker="o",
    label="2025"
)

plt.plot(
    metricas_a26.index,
    metricas_a26["envolvimento"],
    marker="o",
    label="2026"
)

plt.title("Envolvimento - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Quantidade de Envolvimento")

plt.legend()
plt.grid()

plt.show()

Taxa de engajamento

In [ ]:
plt.plot(
    taxa_2025.index,
    taxa_2025,
    marker="o",
    label="2025"
)

plt.plot(
    taxa_2026.index,
    taxa_2026,
    marker="o",
    label="2026"
)

plt.title("Taxa de Engajamento - Meses em Comum")
plt.xlabel("Mês")
plt.ylabel("Taxa de Engajamento (%)")

plt.legend()
plt.grid()

plt.show()

Alcance Orgânico = entre -10% e 10%

Alcance Total = entre 10% e 20%

Curtidas e Reações = entre 20% e 30%

Comentários = entre 20% e 30%

Compartilhamentos = entre 50% e 60%

Cliques = entre 350% e 360%

Salvos = entre 160% e 170%

Impressões Totais = entre 17% e 27%

Envolvimento = entre 35% e 45%

Taxa de Engajamento = entre -45% e -35%